src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -318,22 +331,60 @@ private boolean onRemoveItem() {
         return safeCommand(this, stack, new CmdRemoveItem(msgd, loadout, component, getItem()), msgd);
     }
 
-    private void setupEngineRatingDropDown(boolean aPgiMode) {
+    private <T> boolean addIfEngineExists(final ObservableList<T> aList, final T aValue, final Integer aRating, final EngineType aType, final Faction aFaction) {
+        try {
+            if (aRating != null && aType != null && aFaction != null) {
+                ItemDB.getEngine(aRating, aType, aFaction);
+            }
+            aList.add(aValue);
+            return true;
+        } catch (NoSuchItemException ex) {
+            return false;
+        }
+    }
+
+    private void regenerateEngineRatingDropDown() {
         if (loadout.getChassis() instanceof ChassisStandard) {
             final ChassisStandard chassis = (ChassisStandard) loadout.getChassis();
+
+            final Engine selectedEngine = loadout.getEngine();
+            final Integer selectedRating = selectedEngine != null ? selectedEngine.getRating() : null;
+            final EngineType selectedType = selectedEngine != null ? selectedEngine.getType() : EngineType.STD;
+
             final ObservableList<Integer> items = engineRating.getItems();
             items.clear();
 
-            if (aPgiMode) {
+            if (pgiMode) {
                 for (int r = chassis.getEngineMin(); r <= chassis.getEngineMax(); r += 5) {
-                    items.add(r);
+                    addIfEngineExists(items, r, r, selectedType, chassis.getFaction());
                 }
             }
             else {
                 for (int r = chassis.getEngineMax(); r >= chassis.getEngineMin(); r -= 5) {
-                    items.add(r);
+                    addIfEngineExists(items, r, r, selectedType, chassis.getFaction());
                 }
             }
+
+            boolean previousEngineChangeInProgress = engineChangeInProgress;
+            engineChangeInProgress = true;
+            engineRating.getSelectionModel().select(selectedRating);
```

Comment:
You shouldn't need to do this here as it is already done in `makeEngineGraphic` which is always called before the engine component is rendered with its dropdowns.

Reply:
Done.